### PR TITLE
Added autoreload flag

### PR DESCRIPTION
**Pull request by RomeshA**
This PR adds an optional `autoreload` argument to `run()` so that a local server can be started with automatic reloading of the RPCs. Based on https://gist.github.com/ianschenck/977379a91154fe264897

### CODE REVIEW

**Review by cliffckerr — COMMENTED** [review]
What do you think about this refactoring? I've tried to clean it up a bit, so the app must be run via `app.run()` rather than via the deprecated `sw.run_twisted(app)`, and also made it so you can pass arbitrary arguments to either `flaskapp.run()` or `reactor.run()` -- I've left `autoreload` as-is, but in theory you could also achieve this with `app.run(run_args={'installSignalHandlers':0})`. Thoughts? NB, I haven't tested this at all, I may have made a stupid mistake!

File: scirisweb/sw_app.py [hunk 1]
```diff
@@ -269,18 +304,28 @@ def run(self, with_twisted=True, with_flask=True, with_client=True, do_log=False
...
+            if with_client:
+                twisted_args['client_dir'] = self.config['CLIENT_DIR']
+            if with_flask:
+                twisted_args['flask_app'] = self.flask_app
+            if autoreload:
+                twisted_args['reactor_args']['installSignalHandlers'] = 0
+            run_fcn = lambda: run_twisted(**twisted_args)
+
+        if autoreload:
+            import werkzeug.serving
```
Comment (cliffckerr) [inline scirisweb/sw_app.py]: Any reason why this is imported here rather than at the top with the other werkzeug functions?

Comment (RomeshA) [reply scirisweb/sw_app.py]: No reason apart from that was in the github gist I copied in! I've moved the import now

**Comment by RomeshA** [comment]
Refactoring looks good to me, just in regard to

>  I've left `autoreload` as-is, but in theory you could also achieve this with `app.run(run_args={'installSignalHandlers':0})`. Thoughts? NB, I haven't tested this at all, I may have made a stupid mistake!

I think the main part is wrapping `run_twisted` with `werkzeug.serving.run_with_reloader(run_fcn)`, the `{'installSignalHandlers':0}` is just because having that enabled doesn't work with `run_with_reloader` - but I don't think passing that argument alone to `reactor.run` provides the reloading functionality

**Pull request merged**